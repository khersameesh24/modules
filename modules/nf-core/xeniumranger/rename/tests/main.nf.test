nextflow_process {

    name "Test Process XENIUMRANGER_RENAME"
    script "../main.nf"
    process "XENIUMRANGER_RENAME"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "xeniumranger"
    tag "xeniumranger/rename"
    tag "untar"

    setup {
            run("UNTAR") {
                script "modules/nf-core/untar/main.nf"
                process {
                    """
                    input[0] = [[], file('https://raw.githubusercontent.com/nf-core/test-datasets/spatialxe/Xenium_Prime_Mouse_Ileum_tiny_outs.tar.gz', checkIfExists: true)]
                    """
                }
            }
    }

    test("xeniumranger rename") {
        when {
            process {
                """
                input[0] = UNTAR.out.untar.map{ meta, files -> [ [ id:'test_xeniumranger_resegment' ], file(files)]}
                input[1] = "test_region"
                input[2] = "test_cassette"
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert process.out.bundle != null },
                {
                    assert snapshot(
                        process.out.versions,
                        process.out.bundle.get(0).get(1).findAll { file(it).name !in [
                            'analysis_summary.html',
                            'experiment.xenium',
                        ]}
                    ).match()
                },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'analysis_summary.html' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'experiment.xenium' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'metrics_summary.csv' }).exists() }
            )
        }
    }

    test("xeniumranger rename stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = UNTAR.out.untar.map{ meta, files -> [ [ id:'test_xeniumranger_rename' ], file(files)]}
                input[1] = "test_region"
                input[2] = "test_cassette"
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}