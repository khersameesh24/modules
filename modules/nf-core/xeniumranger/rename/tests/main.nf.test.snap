{
    "xeniumranger rename stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_xeniumranger_rename"
                        },
                        [
                            "fake_file.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,992ecc2b5940913a6f8e6f0e43475761"
                ],
                "bundle": [
                    [
                        {
                            "id": "test_xeniumranger_rename"
                        },
                        [
                            "fake_file.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,992ecc2b5940913a6f8e6f0e43475761"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-01T22:43:21.765405284"
    },
    "xeniumranger rename": {
        "content": [
            [
                "versions.yml:md5,992ecc2b5940913a6f8e6f0e43475761"
            ],
            [
                "/",
                "w",
                "o",
                "r",
                "k",
                "s",
                "p",
                "a",
                "c",
                "e",
                "/",
                "m",
                "o",
                "d",
                "u",
                "l",
                "e",
                "s",
                "/",
                ".",
                "n",
                "f",
                "-",
                "t",
                "e",
                "s",
                "t",
                "/",
                "t",
                "e",
                "s",
                "t",
                "s",
                "/",
                "8",
                "9",
                "b",
                "0",
                "9",
                "c",
                "e",
                "7",
                "5",
                "c",
                "1",
                "6",
                "7",
                "c",
                "d",
                "e",
                "9",
                "4",
                "e",
                "4",
                "f",
                "0",
                "d",
                "4",
                "8",
                "1",
                "6",
                "c",
                "7",
                "d",
                "4",
                "2",
                "/",
                "w",
                "o",
                "r",
                "k",
                "/",
                "3",
                "8",
                "/",
                "8",
                "2",
                "7",
                "0",
                "e",
                "5",
                "2",
                "e",
                "0",
                "8",
                "9",
                "0",
                "8",
                "a",
                "d",
                "9",
                "1",
                "f",
                "a",
                "d",
                "4",
                "f",
                "3",
                "5",
                "8",
                "5",
                "b",
                "c",
                "1",
                "d",
                "/",
                "t",
                "e",
                "s",
                "t",
                "_",
                "x",
                "e",
                "n",
                "i",
                "u",
                "m",
                "r",
                "a",
                "n",
                "g",
                "e",
                "r",
                "_",
                "r",
                "e",
                "n",
                "a",
                "m",
                "e",
                "/",
                "o",
                "u",
                "t",
                "s"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-02T01:00:17.474770445"
    }
}